4주차/[Lv.3]_등굣길/주다현.js

```diff
@@ -0,0 +1,17 @@
+//런타임 에러가 나는 코드랍니다... 아직 해결하지 못했어요ㅠㅠㅠㅠ
+
+function solution(m, n, puddles) {
+  const arr = Array.from(Array(n), () => Array(m).fill(1));
+  puddles.forEach((v) => {
+    const [x, y] = v;
+    arr[x - 1][y - 1] = 0;
+  });
+  for (let i = 1; i < n; i++) {
+    for (let j = 1; j < m; j++) {
+      if (arr[i][j] !== 0) {
+        arr[i][j] = arr[i - 1][j] + arr[i][j - 1];
+      }
```

Comment:
```suggestion
      if (arr[i - 1][j] === null || arr[i][j - 1] === null) {
        arr[i][j] = Math.max(arr[i - 1][j], arr[i][j - 1]);
      } else {
      	arr[i][j] = (arr[i - 1][j] + arr[i][j - 1]) % 1000000007;
      }
```
일반적으로 해당하는 값은 해당하는 값의 위와 왼쪽 값을 합한 것입니다. 하지만 아래 두 가지 경우로 나뉠 수 있습니다.
1. 값의 위와 왼족 값이 하나라도 `물에 잠긴 지역`이라면 `물에 잠긴 지역`을 통과해서 오는 경우는 없으므로 그렇지 않은 경로를 통해서 온 경우만 해당하는 값으로 넣을 수 있습니다. 따라서 두 값 중 즉 `null`과 `정수` 값 중 큰 값은 `정수` 값이므로 그 값을 해당하는 값으로 넣습니다.
2. 위와 같지 않은 경우 해당하는 값의 왼쪽과 위쪽 값을 더한 값을 넣고 [모듈러 산술](https://ko.wikipedia.org/wiki/%EB%AA%A8%EB%93%88%EB%9F%AC_%EC%82%B0%EC%88%A0)을 활용하여 문제에서 지정한 `1000000007`을 나눈 나머지 값을 해당하는 값으로 넣습니다. 문제에서 지정한 값으로 순간마다 나누는 이유는 그렇게 하지 않으면 최종답이 바뀌기 때문이기도 하고, 과정 중 값이 너무 커져 연산을 하지 못해 시간초과 혹은 런타임에러가 날 수 있기 때문입니다.(뇌피셜이어서 따로 또 찾아보시면 좋을 것 같습니다... 🤔)